### PR TITLE
Enables Fortify SCA scans in support of SOC 2 compliance efforts

### PR DESCRIPTION
**Pull request by grimlor**
Note: If the Fortify license is not found in the current user's `.lightbend` directory, the Fortify SCA is not hooked into the build and everything behaves as before.

Changes:
Modifies build.sbt to enable creating NSTs for all modules, outputting to individual folders.
Adds script for scanning all such folders.
Adds the output of said scans for auditing purposes.
Adds scan times CSV for analysis.

References #31403

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -590,18 +590,40 @@ lazy val serialversionRemoverPlugin =
...
+lazy val doesFortifyLicenseExist: Boolean = {
+  import java.nio.file.Files
+  val exists = Files.exists(fortifyLicense.toPath)
+  println(s"doesCredentialExist: ($fortifyLicense) " + exists)
+  exists
+}
+
+def fortifySettings(name: String) =
+  Seq(
+    addCompilerPlugin(("com.lightbend" %% "scala-fortify" % "1.0.21").cross(CrossVersion.patch)),
```
Comment (raboof) [inline build.sbt:609]: (Might be nice to keep the version in `/project/Dependencies.scala`, but not essential)

**Review by raboof — APPROVED** [review]
I'm not too familiar with fortify, but since this change doesn't have any effect unless you have the fortify license configured I'm OK with it.

(I think it would be possible to move this logic out of `build.sbt` into a separate plugin in `/project`, but I'm not sure that really simplifies things, so having it in `build.sbt` seems fine)

File: build.sbt [hunk 2]
```diff
@@ -590,17 +594,36 @@ lazy val serialversionRemoverPlugin =
...
-        Seq("-Xplugin:" + (serialversionRemoverPlugin / Compile / Keys.`package`).value.getAbsolutePath.toString)
+        Seq("-Xplugin:" + (serialversionRemoverPlugin / Compile / Keys.`package`).value.getAbsolutePath)
       else Nil
     ))
 
-def akkaModule(name: String): Project =
+lazy val doesFortifyLicenseExist: Boolean = {
+  import java.nio.file.Files
+  val username = System.getenv("USER")
+  val fortifyLicense = new File(s"/Users/$username/.lightbend/fortify.license")
```
Comment (patriknw) [inline build.sbt:604]: better to use `System.getProperty("user.home")` for finding the home directory

File: fortify-scan.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
```
Comment (patriknw) [inline fortify-scan.sh:1]: can we place this in the `scripts/` directory where we have some other scripts

File: fortify-scan.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  echo Generating report for "$f"
+  BIRTReportGenerator -template "Developer Workbook" -source "$F" -format PDF -output "$scanFolder/$f.pdf"
+done
+
+echo Pushing scan results to repo
+cd fortify-scans || exit
+git add .
+commitMsg="Scan performed on $(date +"%Y-%m-%d_%T")"
+git commit -m "$commitMsg"
+git push origin
```
Comment (patriknw) [inline fortify-scan.sh:38]: Should this really push to the repo? What is the process for running these scripts? Are they supposed to be run by CI or is it manually from a local machine?

Comment (grimlor) [reply fortify-scan.sh:38]: Right now, it's manual but I am writing this with the idea of automation. Right now, it's manual but I am writing this with the idea of automation, so CI is the goal. 

As for pushing to the repo, I've changed the script to push the results to a private repo, lightbend/fortify-scans under a folder named for the project being scanned. Technically, anyone could have Fortify SCA and run the same scans but I figured it best to not expose our vulnerabilities so directly as to include with the project.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by grimlor** [comment]
I'm not yet a project collaborator so the above checks have to be run by one of y'all. :)

**Comment by patriknw** [comment]
There is a formatting issue in build.sbt. Could you run `scalafmtSbt` and commit the changes.

If it's ready after that we can merge it.

**Comment by johanandren** [comment]
Trying to re-trigger validation

**Pull request merged**